File: high_load.go
package main

import (
    "fmt"
    "time"
)

func multiply(a, b [][]int) [][]int {
    n := len(a)
    c := make([][]int, n)
    for i := range c {
        c[i] = make([]int, n)
    }

    for i := 0; i < n; i++ {
        for j := 0; j < n; j++ {
            sum := 0
            for k := 0; k < n; k++ {
                sum += a[i][k] * b[k][j]
            }
            c[i][j] = sum
        }
    }
    return c
}

func main() {
    const n = 2000

    // Initialize matrices
    a := make([][]int, n)
    b := make([][]int, n)
    for i := 0; i < n; i++ {
        a[i] = make([]int, n)
        b[i] = make([]int, n)
        for j := 0; j < n; j++ {
            a[i][j] = (i + j) % 10
            b[i][j] = (i * j) % 10
        }
    }

    fmt.Println("Program starting...")

    start := time.Now()
    fmt.Printf("Start time: %.6f\n", float64(start.UnixNano())/1e9)

    c := multiply(a, b)

    end := time.Now()
    fmt.Printf("End time: %.6f\n", float64(end.UnixNano())/1e9)

    elapsed := end.Sub(start).Seconds()
    fmt.Printf("Time elapsed: %.8f seconds\n", elapsed)

    fmt.Printf("c[0][0] = %d\n", c[0][0])
}

File: high_load.rs
use std::time::{SystemTime, UNIX_EPOCH};

fn mat_multiply(a: &Vec<Vec<i32>>, b: &Vec<Vec<i32>>, size: usize) -> Vec<Vec<i32>> {
    let mut result = vec![vec![0; size]; size];

    for i in 0..size {
        for j in 0..size {
            let mut sum = 0;
            for k in 0..size {
                sum += a[i][k] * b[k][j];
            }
            result[i][j] = sum;
        }
    }
    result
}

fn main() {
    const N: usize = 2000;

    // Initialize matrices
    let mut a = vec![vec![0; N]; N];
    let mut b = vec![vec![0; N]; N];

    for i in 0..N {
        for j in 0..N {
            a[i][j] = ((i + j) % 10) as i32;
            b[i][j] = ((i * j) % 10) as i32;
        }
    }

    println!("Program starting...");

    let start = SystemTime::now();
    let start_secs = start.duration_since(UNIX_EPOCH).unwrap().as_secs_f64();
    println!("Start time: {:.6}", start_secs);

    let c = mat_multiply(&a, &b, N);

    let end = SystemTime::now();
    let end_secs = end.duration_since(UNIX_EPOCH).unwrap().as_secs_f64();
    println!("End time: {:.6}", end_secs);

    let elapsed = end.duration_since(start).unwrap().as_secs_f64();
    println!("Time elapsed: {:.8} seconds", elapsed);

    println!("c[0][0] = {}", c[0][0]);
}
